consensus/core/src/dag_state.rs

```diff
@@ -246,6 +246,25 @@ impl DagState {
             );
         }
 
+        // Initialize the scoring metrics according to the stored blocks
+        let threshold_clock_round = state.threshold_clock_round();
+        for index in 0..num_authorities {
+            let authority_index = state.context.committee.to_authority_index(index).unwrap();
+            let eviction_round = state.evicted_rounds[index];
+            let hostname = &state.context.committee.authority(authority_index).hostname;
+            let stored_block_rounds = state
+                .store
+                .scan_block_rounds_by_author(authority_index)
+                .expect("Database error");
+            state.context.metrics.load_scoring_metrics_from_storage(
+                authority_index,
+                hostname,
+                stored_block_rounds,
+                threshold_clock_round,
+                eviction_round,
+            );
+        }
```

Comment:
```suggestion
        for (authority_index, _) in state.context.committee.authorities() {
            let eviction_round = state.evicted_rounds[authority_index];
            let hostname = &state.context.committee.authority(authority_index).hostname;
            let stored_block_rounds = state
                .store
                .scan_block_rounds_by_author(authority_index)
                .expect("Database error");
            state.context.metrics.load_scoring_metrics_from_storage(
                authority_index,
                hostname,
                stored_block_rounds,
                threshold_clock_round,
                eviction_round,
            );
        }
```

Reply:
Actually, you can also use the authority itself in the iterator to get the hostname too.
```suggestion
        for (authority_index, authority) in state.context.committee.authorities() {
            let eviction_round = state.evicted_rounds[authority_index];
            let hostname = &authority.hostname;
            let stored_block_rounds = state
                .store
                .scan_block_rounds_by_author(authority_index)
                .expect("Database error");
            state.context.metrics.load_scoring_metrics_from_storage(
                authority_index,
                hostname,
                stored_block_rounds,
                threshold_clock_round,
                eviction_round,
            );
        }
```

Reply:
done!